src/components/common/EthHashInfo/index.tsx

```diff
@@ -1,104 +1,23 @@
-import { ethers } from 'ethers'
+import { EthHashInfo } from '@safe-global/safe-react-components'
+import { getBlockExplorerLink } from '../../../utils/chains'
+import { useCurrentChain } from '../../../hooks/useChains'
 
-type EthHashInfoProps = {
-  address: string
-  chainId?: string
-  name?: string | null
-  showAvatar?: boolean
-  showCopyButton?: boolean
-  prefix?: string
-  showPrefix?: boolean
-  copyPrefix?: boolean
-  shortAddress?: boolean
-  customAvatar?: string
-  hasExplorer?: boolean
-  avatarSize?: number
-  children?: React.ReactNode
-}
-
-const EthHashInfo = ({
-  address,
-  customAvatar,
-  prefix = '',
-  copyPrefix,
-  showPrefix,
-  shortAddress = true,
-  showAvatar = true,
-  avatarSize,
-  name,
-  showCopyButton,
-  hasExplorer,
-
-  children,
-}: EthHashInfoProps): ReactElement => {
-  const [fallbackToIdenticon, setFallbackToIdenticon] = useState(false)
-  const shouldPrefix = ethers.utils.isAddress(address)
-
-  return (
-    <div className={css.container}>
-      {showAvatar && (
-        <div className={classnames(css.avatar, { [css.resizeAvatar]: !avatarSize })}>
-          {!fallbackToIdenticon && customAvatar ? (
-            <img
-              src={customAvatar}
-              alt={address}
-              onError={() => setFallbackToIdenticon(true)}
-              width={avatarSize}
-              height={avatarSize}
-            />
-          ) : (
-            <Identicon address={address} size={avatarSize} />
-          )}
-        </div>
-      )}
-
-      <div className={css.nameRow}>
-        {name && (
-          <Typography variant="body2" component="div" textOverflow="ellipsis" overflow="hidden" title={name}>
-            {name}
-          </Typography>
-        )}
-
-        <Box className={css.addressRow}>
-          <Typography fontWeight="inherit" fontSize="inherit">
-            {showPrefix && shouldPrefix && prefix && <b>{prefix}:</b>}
-            <span className={css.mobileAddress}>{shortenAddress(address)}</span>
-            <span className={css.desktopAddress}>{shortAddress ? shortenAddress(address) : address}</span>
-          </Typography>
-
-          {showCopyButton && (
-            <CopyAddressButton prefix={prefix} address={address} copyPrefix={shouldPrefix && copyPrefix} />
-          )}
-
-          {hasExplorer && <ExplorerLink address={address} />}
-          {children}
-        </Box>
-      </div>
-    </div>
-  )
-}
+import type { EthHashInfoProps } from '@safe-global/safe-react-components'
 
 const PrefixedEthHashInfo = ({
   showName = true,
   ...props
 }: EthHashInfoProps & { showName?: boolean }): ReactElement => {
   const settings = useAppSelector(selectSettings)
-  const currentChainId = useChainId()
-  const chain = useAppSelector((state) => selectChainById(state, props.chainId || currentChainId))
+  const currentChain = useCurrentChain()
+  const chain = useAppSelector((state) => selectChainById(state, props.chainId || currentChain?.chainId || ''))
```

Comment:
We should fall back to `useChainId` as it returns a default chain.

Reply:
I'd keep the logic as before but call `getBlockExplorerLink` with `chain`.

Reply:
aaah :D

Reply:
make sense no doubt 😂

Reply:
done